flows/boundary.py

```diff
@@ -806,6 +808,7 @@ def __call__(
         grouped_concepts: dict[TextBlockId, list[VespaConcept]],
         response: VespaResponse,
         data_id: VespaDataId,
+        logger: Union[Logger, LoggingAdapter],
```

Comment:
nitpick (ignore this if you like): is it a good idea to have a function that can _only_ run if you pass in a logger? Could this ever hinder testing or reuse of this function, if reuse was likely?

I think reuse out of this context is unlikely but an alternative here would be to pass an optional logger. This depends on whether you think logging is a core part of the method's reason for being (ie it should **never** not log) I guess...

Reply:
Thanks for the comment! I would rather enforce the logging parameter to avoid conditional paths within the function!